src/Component/Pages/PostPage/PaginatedPosts.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+import Post from "./Post";
+import PaginationRounded from "./PaginationRounded";
+
+interface IPost {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+const PaginatedPosts: React.FC = () => {
+  const [posts, setPosts] = useState<IPost[]>([]);
+  const [currentPageNo, setCurrentPageNo] = useState(1);
+  const postsPerPage = 10;
+  const [loadingData, setLoadingData] = useState(false);
+  const [error, setError] = useState<null | string>(null);
+
+  useEffect(() => {
+    setLoadingData(true);
+    fetch("https://jsonplaceholder.typicode.com/posts")
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error("Network Response was not ok");
+        }
+        console.log(res);
+        return res.json();
+      })
+      .then((data) => {
+        setPosts(data);
+        setLoadingData(false);
+      })
+      .catch((error) => {
+        setError(error.message); //capture and store the error message
+        console.log(error.message);
+      });
+  }, []);
+
+  const handleUpdatePage = (inputValue: number) => {
+    setCurrentPageNo(inputValue);
+  };
+  //Get current posts
+  const indexOfLastPost = currentPageNo * postsPerPage;
+  const indexOfFirstPost = indexOfLastPost - postsPerPage;
+  const currentPosts = posts.slice(indexOfFirstPost, indexOfLastPost);
```

Comment:
It's better to use ```useMemo``` when you have a complicated variable. Don't forget to depend it to all variables you used inside of it

Reply:
Used useMemo in PaginationRounded component.